### PR TITLE
Add children val to ViewGroup which returns a Sequence<View>

### PR DESCRIPTION
**Pull request by panpanini**
References #138

Adds a `children` val extension to ViewGroup which returns a Sequence of child views

I'm not too sure about the naming of the test method, so if someone has a better suggestion :pray:

### CODE REVIEW

File: src/androidTest/java/androidx/view/ViewGroupTest.kt [hunk 1]
```diff
@@ -199,6 +199,18 @@ class ViewGroupTest {
         }
     }
 
+    @Test fun children() {
+        val views = listOf(View(context), View(context), View(context))
+        views.forEach { viewGroup.addView(it) }
+
+        var index = 0
```
Comment (JakeWharton) [inline src/androidTest/java/androidx/view/ViewGroupTest.kt]: use `forEachIndexed`

File: src/androidTest/java/androidx/view/ViewGroupTest.kt [hunk 2]
```diff
@@ -199,6 +199,18 @@ class ViewGroupTest {
...
+        viewGroup
+                .children
+                .forEach { child ->
```
Comment (JakeWharton) [inline src/androidTest/java/androidx/view/ViewGroupTest.kt]: You can put all these on the same line as `viewGroup`

File: src/main/java/androidx/view/ViewGroup.kt [hunk 3]
```diff
@@ -71,6 +71,16 @@ operator fun ViewGroup.iterator() = object : MutableIterator<View> {
     override fun remove() = removeViewAt(--index)
 }
 
+/** Returns a [Sequence] over the child views in this view group. */
+val ViewGroup.children: Sequence<View>
+    get() {
+        var index = -1
+        return generateSequence {
```
Comment (JakeWharton) [inline src/main/java/androidx/view/ViewGroup.kt]: This has quite a bit of overhead. Why don't we just return an `object : Sequence` whose function delegates to our existing `iterator()` method instead.

Comment (panpanini) [reply src/main/java/androidx/view/ViewGroup.kt]: would this be the same as something like:
``` kotlin
val ViewGroup.children: Sequence<View>
  get() = iterator().asSequence()
```

Comment (JakeWharton) [reply src/main/java/androidx/view/ViewGroup.kt]: I'd need to check what `asSequence` does. It probably prevents multiple iteration which my proposed solution doesn't.

Comment (panpanini) [reply src/main/java/androidx/view/ViewGroup.kt]: yeah it does, thats expected for a Sequence right?

Comment (JakeWharton) [reply src/main/java/androidx/view/ViewGroup.kt]: It's not required, no.

File: src/androidTest/java/androidx/view/ViewGroupTest.kt [hunk 4]
```diff
@@ -199,6 +199,15 @@ class ViewGroupTest {
         }
     }
 
+    @Test fun children() {
+        val views = listOf(View(context), View(context), View(context))
+        views.forEach { viewGroup.addView(it) }
+
+        viewGroup.children.forEachIndexed { index, child ->
+                    assertSame(views[index], child)
```
Comment (romainguy) [inline src/androidTest/java/androidx/view/ViewGroupTest.kt]: Your indentation is incorrect FYI

**Comment by JakeWharton** [comment]
You still need to run the `updateApi` task and commit its changes.

**Comment by panpanini** [comment]
ah whoops, I ran the task but I'm not sure if I committed the results. I'll check that now

**Comment by JakeWharton** [comment]
Squashed and merged. Thanks!

**Pull request closed without merging**